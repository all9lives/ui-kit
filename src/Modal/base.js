import { PureComponent, cloneElement } from 'react'

const changeState = (state, value) => () => ({
  [state]: value
})

const TRANSITION_TIME = 300

export class StateProvider extends PureComponent {
  constructor (props) {
    super(props)
    this.state = {
      isOpen: props.isOpen,
      isOpening: false
    }
  }
  componentWillReceiveProps (nextProps) {
    if (this.props.isOpen !== nextProps.isOpen) {
      let first = null
      let next = null
      if (nextProps.isOpen) {
        first = changeState('isOpen', nextProps.isOpen)
        next = changeState('isOpening', nextProps.isOpen)
      } else {
        first = changeState('isOpening', nextProps.isOpen)
        next = changeState('isOpen', nextProps.isOpen)
      }
      this.setState(first, () => {
        setTimeout(() => {
          this.setState(next)
        }, TRANSITION_TIME)
      })
    }
  }
  render () {
    const { children, onRequestClose } = this.props
    const { isOpen, isOpening } = this.state
    if (!isOpen) {
      return null
    }
    return cloneElement(children, { isOpen, isOpening, onRequestClose })
  }
}
