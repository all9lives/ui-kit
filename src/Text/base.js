import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Size from '../utils/Size'
import Color from '../utils/Color'

const base = css`
  font-size: ${props => Unit.convert(Size[props.size] || props.size)};
  color: ${props => (Color[props.color] || props.color)};
  margin: ${props => Unit.parse(props.margin, (margin) => margin * Size.spacing)};
  align-self: ${props => props.alignSelf};
`

export default base
