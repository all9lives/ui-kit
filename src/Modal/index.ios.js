import Modal from './index.android'

export default Modal
