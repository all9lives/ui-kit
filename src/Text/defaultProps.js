import Color from '../utils/Color'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export default {
  size: Unit.convert(Size.regular),
  color: Color.black
}
