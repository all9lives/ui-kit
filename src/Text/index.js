import React from 'react';
import { Text as PrimitiveText } from 'react-primitives';

const Text = ({ children }) => (
  <PrimitiveText>{children}</PrimitiveText>
);

export default Text;
