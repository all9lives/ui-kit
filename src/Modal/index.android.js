import styled from 'styled-components/native'
import defaultProps from './defaultProps'

const Modal = styled.Modal.attrs({
  visible: props => props.isOpen
})`
`
Modal.defaultProps = defaultProps

export default Modal
