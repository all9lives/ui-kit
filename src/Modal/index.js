import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { StateProvider } from './base'
import defaultProps from './defaultProps'
import View from '../View'

export const StyledModal = styled(View)`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  opacity: ${props => props.isOpening ? 1 : 0};
  transition: opacity 300ms ease-in;
`
StyledModal.defaultProps = defaultProps

export default class Modal extends PureComponent {
  render () {
    const { children } = this.props
    return (
      <StateProvider {...this.props}>
        <StyledModal>
          {children}
        </StyledModal>
      </StateProvider>
    )
  }
}
