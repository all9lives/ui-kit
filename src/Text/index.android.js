import styled from 'styled-components/native'
import base from './base'
import defaultProps from './defaultProps'
import Size from '../utils/Size'

const getLineHeight = (props) => {
  if (props.content) {
    return Size.spacing * 2.5
  }
  if (props.size === 'title') {
    return Size.spacing * 3
  }
  return Size.spacing * 2
}

const Text = styled.Text`
  ${base}
  line-height: ${props => getLineHeight(props)};
`
Text.defaultProps = defaultProps

export default Text
