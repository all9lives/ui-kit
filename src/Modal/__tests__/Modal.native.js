/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Modal from '../index.ios.js'
import Text from '../../Text'

describe('Modal.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Modal><Text>Content</Text></Modal>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
