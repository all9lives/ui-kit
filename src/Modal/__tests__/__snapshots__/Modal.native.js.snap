// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal.native renders a snapshot 1`] = `
<Component
  hardwareAccelerated={false}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  visible={true}
>
  <Styled(Text)
    alignSelf="auto"
    color="#282828"
    margin={0}
    size={16}
  >
    Content
  </Styled(Text)>
</Component>
`;
